File: lab3/FileReverse.c
/*
* Vincent Titterton
* vtittert
* 12M
* 1/27/2019
* Takes a file as input and creates a file with each
* word from the input reversed and on a new line
*/

#include<stdio.h>
#include<stdlib.h>
#include <string.h>

void stringReverse(char* s)
{
    int i = 0;
    int j = strlen(s) - 1;
    while (j - i > 0)
    {
        char temp = s[i];
        s[i] = s[j];
        s[j] = temp;
        i++;
        j--;
    }
}


int main(int argc, char* argv[])
{
	FILE* in; /* file handle for input */
 	FILE* out; /* file handle for output */
 	char word[256]; /* char array to store words from input file */
 /* check command line for correct number of arguments */
	if( argc != 3 )
	{
		printf("Usage: %s <input file> <output file>\n", argv[0]);
		exit(EXIT_FAILURE);
	}
 /* open input file for reading */
	in = fopen(argv[1], "r");
	if( in==NULL )
	{
		printf("Unable to read from file %s\n", argv[1]);
		exit(EXIT_FAILURE);
	}
 /* open output file for writing */
	out = fopen(argv[2], "w");
	if( out==NULL )
	{
		printf("Unable to write to file %s\n", argv[2]);
		exit(EXIT_FAILURE);
	}
 /* read words from input file, print on separate lines to output file*/
	while( fscanf(in, " %s", word) != EOF )
	{
        stringReverse(word);
		fprintf(out, "%s\n", word);
	}
 /* close input and output files */
	fclose(in);
	fclose(out);
	return(EXIT_SUCCESS);
}

File: lab2/FileReverse.java
/*
 * Vincent Titterton
 * vtittert
 * 12M
 * 1/24/2019
 * This program takes an input file, separates it into tokens, 
 * and creates a new file listing the reverse of each token
 */

import java.io.*;
import java.util.Scanner;

public class FileReverse {
	public static void main(String[] args) throws IOException
	{
		Scanner in = null;
		PrintWriter out = null;
		String line = null;
		String[] token = null;
		int i, n, lineNumber = 0;
		// check number of command line arguments is at least 2
		if (args.length != 2) {
			System.err.println("Usage: FileCopy <input file> <output file>");
			System.exit(1);
		}
		// open files
		try
		{
			new Scanner(new File(args[0]));
		}
		catch (Exception e)
		{
			System.err.println(args[0] + " (No such file or directory)");
			System.err.println("Usage: FileCopy <input file> <output file>");
			System.exit(1);
		}
		in = new Scanner(new File(args[0]));
		out = new PrintWriter(new FileWriter(args[1]));
		// read lines from in, extract and print tokens from each line
		while (in.hasNextLine()) {
			lineNumber++;
			// trim leading and trailing spaces, then add one trailing space so
			// split works on blank lines
			line = in.nextLine().trim() + " ";
			// split line around white space
			token = line.split("\\s+");
			// print out tokens
			n = token.length;
			for (i = 0; i < n; i++) {
				out.println(" " + stringReverse(token[i], token[i].length()));
			}
		}
		// close files
		in.close();
		out.close();
	}
	
	public static String stringReverse(String s, int n)
	{
		if(n > 1)
		{
			return s.substring(n - 1, n) + stringReverse(s.substring(0, n - 1), n - 1);
		}
		return s;
	}
}

